pkg/controller/vpaevictionrequirements/reconciler.go

```diff
@@ -0,0 +1,155 @@
+	splitWindowAnnotation := strings.Split(windowAnnotation, ",")
+	if len(splitWindowAnnotation) != 2 {
+		err := fmt.Errorf("error during parsing the maintenance window from annotation. Value is not in format '<begin>,<end>': %s", windowAnnotation)
+		log.Error(err, "Error during reconciling for downscaling in maintenance only")
+		// No need to retry reconciling this VPA until it has been updated with a fixed annotation, therefore not returning the error
+		return reconcile.Result{}, nil
+	}
+
+	maintenanceTimeWindow, err := timewindow.ParseMaintenanceTimeWindow(splitWindowAnnotation[0], splitWindowAnnotation[1])
+	if err != nil {
+		log.Error(err, "Error during parsing the maintenance window from start and end time", "begin", splitWindowAnnotation[0], "end", splitWindowAnnotation[1])
+		// No need to retry reconciling this VPA until it has been updated with a fixed annotation, therefore not returning the error
+		return reconcile.Result{}, nil
+	}
+
+	if isNowInMaintenanceTimeWindow := maintenanceTimeWindow.Contains(r.Clock.Now()); isNowInMaintenanceTimeWindow {
+		log.Info("Shoot is inside maintenance window, removing the EvictionRequirement to allow downscaling", "maintenanceWindow", maintenanceTimeWindow)
+
+		if _, err := controllerutil.CreateOrUpdate(ctx, r.SeedClient, vpa, func() error {
+			vpa.Spec.UpdatePolicy.EvictionRequirements = nil
+			return nil
+		}); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		// requeue when the maintenance window ends, such that we can add the EvictionRequirement again
+		endTime := maintenanceTimeWindow.AdjustedEnd(r.Clock.Now())
+		requeueAfter := endTime.Sub(r.Clock.Now())
+		log.Info("Requeueing to the end of the maintenance window", "requeueAfter", requeueAfter)
+		return reconcile.Result{RequeueAfter: requeueAfter}, nil
+	}
+
+	log.Info("Shoot is not inside maintenance window, adding EvictionRequirement to deny downscaling", "maintenanceWindow", maintenanceTimeWindow)
+
+	if _, err = controllerutil.CreateOrUpdate(ctx, r.SeedClient, vpa, func() error {
+		vpa.Spec.UpdatePolicy.EvictionRequirements = upscaleOnlyRequirement
+		return nil
+	}); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// requeue when the next maintenance window begins, such that we can remove the EvictionRequirement
+	nextWindowBegin := maintenanceTimeWindow.AdjustedBegin(r.Clock.Now())
+	if nextWindowBegin.Before(r.Clock.Now()) {
+		nextWindowBegin = nextWindowBegin.AddDate(0, 0, 1)
+	}
+	requeueAfter := nextWindowBegin.Sub(r.Clock.Now())
+	log.Info("Requeueing to the begin of the next maintenance window", "requeueAfter", requeueAfter)
+	return reconcile.Result{RequeueAfter: requeueAfter}, nil
+}
+
+func (r *Reconciler) reconcileVPAForDownscaleDisabled(ctx context.Context, log logr.Logger, vpa *vpaautoscalingv1.VerticalPodAutoscaler) (reconcile.Result, error) {
+	log.Info("Adding EvictionRequirement for vpa to deny downscaling")
```

Comment:
For consistency reasons with other logs:

```suggestion
	log.Info("Adding EvictionRequirement for VPA to deny downscaling")
```

Reply:
done.